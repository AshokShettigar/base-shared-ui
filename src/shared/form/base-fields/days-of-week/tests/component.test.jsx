import { shallow } from 'enzyme';

import DaysOfWeek from '../component';

describe('DaysOfWeek: ', () => {
    describe('component', () => {
        const requiredProps = {
            onChange: jest.fn()
        };

        it('should render component with default props', () => {
            const wrapper = shallow(<DaysOfWeek {...requiredProps} />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('should render component with id prop', () => {
            const wrapper = shallow(<DaysOfWeek {...requiredProps} id="id" />);

            expect(wrapper).toMatchSnapshot();
        });

        it('should render component with label prop', () => {
            const wrapper = shallow(<DaysOfWeek {...requiredProps} label="label" />);

            expect(wrapper).toMatchSnapshot();
        });

        it('should render component with name prop', () => {
            const wrapper = shallow(<DaysOfWeek {...requiredProps} name="name" />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('should render component with selected days', () => {
            const wrapper = shallow(<DaysOfWeek {...requiredProps} value={[1, 2, 3]} />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('should render component with full names of days', () => {
            const wrapper = shallow(<DaysOfWeek {...requiredProps} format="long" />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('should render component with required prop', () => {
            const wrapper = shallow(<DaysOfWeek {...requiredProps} isRequired />);

            expect(wrapper).toMatchSnapshot();
        });

        it('should render component with isDisabled prop', () => {
            const wrapper = shallow(<DaysOfWeek {...requiredProps} isDisabled />).dive();

            expect(wrapper).toMatchSnapshot();
        });

        it('should render component with class names props', () => {
            const wrapper = shallow(<DaysOfWeek
                {...requiredProps}
                className="className"
                labelClassName="labelClassName"
                cellClassName="cellClassName"
            />).dive();

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('method', () => {
        it('trigger handleChange on selected cell', () => {
            const props = {
                name: 'name',
                value: [1, 2, 3]
            };
            const selectedDayOfWeek = 4;
            const onChange = jest.fn();
            const selectedDaysOfWeek = [...props.value, selectedDayOfWeek];

            const wrapper = shallow(<DaysOfWeek {...props} onChange={onChange} />).dive();

            wrapper.instance().handleChange(selectedDayOfWeek);

            expect(onChange).toBeCalledWith(props.name, selectedDaysOfWeek);
        });

        it('trigger handleChange on unselected cell', () => {
            const props = {
                name: 'name',
                value: [1, 2, 3, 4]
            };
            const selectedDayOfWeek = 4;
            const onChange = jest.fn();
            const selectedDaysOfWeek = props.value.filter(day => day !== selectedDayOfWeek);

            const wrapper = shallow(<DaysOfWeek {...props} onChange={onChange} />).dive();

            wrapper.instance().handleChange(selectedDayOfWeek);

            expect(onChange).toBeCalledWith(props.name, selectedDaysOfWeek);
        });
    });
});
