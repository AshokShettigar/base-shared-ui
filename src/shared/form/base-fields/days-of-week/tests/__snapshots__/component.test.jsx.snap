// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DaysOfWeek:  component should render component with class names props 1`] = `
<div
  className="root className"
  data-component="DaysOfWeek"
>
  <WithRender
    error={null}
    hidden={false}
    rendered={true}
  >
    <WithRender
      className="labelClassName"
      hidden={false}
      rendered={true}
      required={false}
    >
      <div
        className="daysList"
      >
        <DayCell
          className="dayCell cellClassName"
          isDisabled={false}
          label="Sun"
          onChange={[Function]}
          value={0}
        />
        <DayCell
          className="dayCell cellClassName"
          isDisabled={false}
          label="Mon"
          onChange={[Function]}
          value={1}
        />
        <DayCell
          className="dayCell cellClassName"
          isDisabled={false}
          label="Tue"
          onChange={[Function]}
          value={2}
        />
        <DayCell
          className="dayCell cellClassName"
          isDisabled={false}
          label="Wed"
          onChange={[Function]}
          value={3}
        />
        <DayCell
          className="dayCell cellClassName"
          isDisabled={false}
          label="Thu"
          onChange={[Function]}
          value={4}
        />
        <DayCell
          className="dayCell cellClassName"
          isDisabled={false}
          label="Fri"
          onChange={[Function]}
          value={5}
        />
        <DayCell
          className="dayCell cellClassName"
          isDisabled={false}
          label="Sat"
          onChange={[Function]}
          value={6}
        />
      </div>
    </WithRender>
  </WithRender>
</div>
`;

exports[`DaysOfWeek:  component should render component with default props 1`] = `
<div
  className="root"
  data-component="DaysOfWeek"
>
  <WithRender
    error={null}
    hidden={false}
    rendered={true}
  >
    <WithRender
      hidden={false}
      rendered={true}
      required={false}
    >
      <div
        className="daysList"
      >
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Sun"
          onChange={[Function]}
          value={0}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Mon"
          onChange={[Function]}
          value={1}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Tue"
          onChange={[Function]}
          value={2}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Wed"
          onChange={[Function]}
          value={3}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Thu"
          onChange={[Function]}
          value={4}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Fri"
          onChange={[Function]}
          value={5}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Sat"
          onChange={[Function]}
          value={6}
        />
      </div>
    </WithRender>
  </WithRender>
</div>
`;

exports[`DaysOfWeek:  component should render component with full names of days 1`] = `
<div
  className="root"
  data-component="DaysOfWeek"
>
  <WithRender
    error={null}
    hidden={false}
    rendered={true}
  >
    <WithRender
      hidden={false}
      rendered={true}
      required={false}
    >
      <div
        className="daysList"
      >
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Sunday"
          onChange={[Function]}
          value={0}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Monday"
          onChange={[Function]}
          value={1}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Tuesday"
          onChange={[Function]}
          value={2}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Wednesday"
          onChange={[Function]}
          value={3}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Thursday"
          onChange={[Function]}
          value={4}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Friday"
          onChange={[Function]}
          value={5}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Saturday"
          onChange={[Function]}
          value={6}
        />
      </div>
    </WithRender>
  </WithRender>
</div>
`;

exports[`DaysOfWeek:  component should render component with id prop 1`] = `
<DaysOfWeek
  error={null}
  format="short"
  hidden={false}
  id="id"
  isDisabled={false}
  isRequired={false}
  onChange={[Function]}
  value={Array []}
/>
`;

exports[`DaysOfWeek:  component should render component with isDisabled prop 1`] = `
<div
  className="root"
  data-component="DaysOfWeek"
>
  <WithRender
    error={null}
    hidden={false}
    rendered={true}
  >
    <WithRender
      hidden={false}
      rendered={true}
      required={false}
    >
      <div
        className="daysList"
      >
        <DayCell
          className="dayCell"
          isDisabled={true}
          label="Sun"
          onChange={[Function]}
          value={0}
        />
        <DayCell
          className="dayCell"
          isDisabled={true}
          label="Mon"
          onChange={[Function]}
          value={1}
        />
        <DayCell
          className="dayCell"
          isDisabled={true}
          label="Tue"
          onChange={[Function]}
          value={2}
        />
        <DayCell
          className="dayCell"
          isDisabled={true}
          label="Wed"
          onChange={[Function]}
          value={3}
        />
        <DayCell
          className="dayCell"
          isDisabled={true}
          label="Thu"
          onChange={[Function]}
          value={4}
        />
        <DayCell
          className="dayCell"
          isDisabled={true}
          label="Fri"
          onChange={[Function]}
          value={5}
        />
        <DayCell
          className="dayCell"
          isDisabled={true}
          label="Sat"
          onChange={[Function]}
          value={6}
        />
      </div>
    </WithRender>
  </WithRender>
</div>
`;

exports[`DaysOfWeek:  component should render component with label prop 1`] = `
<DaysOfWeek
  error={null}
  format="short"
  hidden={false}
  isDisabled={false}
  isRequired={false}
  label="label"
  onChange={[Function]}
  value={Array []}
/>
`;

exports[`DaysOfWeek:  component should render component with name prop 1`] = `
<div
  className="root"
  data-component="DaysOfWeek"
>
  <WithRender
    error={null}
    hidden={false}
    rendered={true}
  >
    <WithRender
      hidden={false}
      rendered={true}
      required={false}
    >
      <div
        className="daysList"
      >
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Sun"
          onChange={[Function]}
          value={0}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Mon"
          onChange={[Function]}
          value={1}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Tue"
          onChange={[Function]}
          value={2}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Wed"
          onChange={[Function]}
          value={3}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Thu"
          onChange={[Function]}
          value={4}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Fri"
          onChange={[Function]}
          value={5}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Sat"
          onChange={[Function]}
          value={6}
        />
      </div>
    </WithRender>
  </WithRender>
</div>
`;

exports[`DaysOfWeek:  component should render component with required prop 1`] = `
<DaysOfWeek
  error={null}
  format="short"
  hidden={false}
  isDisabled={false}
  isRequired={true}
  onChange={[Function]}
  value={Array []}
/>
`;

exports[`DaysOfWeek:  component should render component with selected days 1`] = `
<div
  className="root"
  data-component="DaysOfWeek"
>
  <WithRender
    error={null}
    hidden={false}
    rendered={true}
  >
    <WithRender
      hidden={false}
      rendered={true}
      required={false}
    >
      <div
        className="daysList"
      >
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Sun"
          onChange={[Function]}
          value={0}
        />
        <DayCell
          className="dayCell selected"
          isDisabled={false}
          label="Mon"
          onChange={[Function]}
          value={1}
        />
        <DayCell
          className="dayCell selected"
          isDisabled={false}
          label="Tue"
          onChange={[Function]}
          value={2}
        />
        <DayCell
          className="dayCell selected"
          isDisabled={false}
          label="Wed"
          onChange={[Function]}
          value={3}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Thu"
          onChange={[Function]}
          value={4}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Fri"
          onChange={[Function]}
          value={5}
        />
        <DayCell
          className="dayCell"
          isDisabled={false}
          label="Sat"
          onChange={[Function]}
          value={6}
        />
      </div>
    </WithRender>
  </WithRender>
</div>
`;
